db/migrate/20171012130125_create_node_attributes.rb

```diff
@@ -0,0 +1,43 @@
+class CreateNodeAttributes < ActiveRecord::Migration[5.0]
+  include SearchPathHelpers
+
+  def up
+    with_search_path('revamp') do
+      create_table :node_attributes do |t|
+        t.references :node, null: false, foreign_key: true, index: true
+        t.references :attribute, null: false, foreign_key: true, index: true
+        t.timestamps
+      end
+
+      add_index :node_attributes, [:node_id, :attribute_id], unique: true
+
+      create_table :node_attributes_double_values do |t|
+        t.references :node_attribute, null: false, foreign_key: true, index: true
+        t.integer :year
+        t.column :value, 'double precision', null: false
+        t.timestamps
+      end
+
+      add_index :node_attributes_double_values, [:node_attribute_id, :year], unique: true,
+        name: :index_node_attributes_double_values_on_id_and_year
```

Comment:
we may need a context_id in here somewhere, to account for the fact that a node having different attribute values for different contexts. Alternatively, we agree to not do that, but either way, I think it's something we should discuss